python/ray/tune/tests/test_multinode_sync.py

```diff
@@ -161,6 +165,143 @@ def on_step_begin(self, iteration, trials, **info):
             local_dir="/tmp/ray_results/",
         )
 
+    def testCheckpointSync(self):
```

Comment:
I guess it's also good to document in what sense this test asserts sync to driver behavior.
From what I can tell, it only directly looks into the content of trial checkpoint on each node. It does not directly assert the content on driver node. It's sort of based on the rationale that if the content on driver node is not correct, then we won't be able to start each trial from step = 10 and as a result the final checkpoint at the end of 2nd run won't be 18 and 19.

This is kind of hard to follow. Especially in the context of mapping and mount of docker node directory onto host machine. Would be good to add some clarification.

Reply:
Added a long docstring to explain test behavior